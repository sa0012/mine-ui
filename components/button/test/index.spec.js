import { mount } from '@vue/test-utils'
import Button from '..'

test('size', () => {
  const wrapper = mount(Button, {
    propsData: {
      size: 'large'
    }
  })
  expect(wrapper).toMatchSnapshot()
})

test('click event', () => {
  const onClick = jest.fn()
  const wrapper = mount(Button, {
    context: {
      on: {
        click: onClick
      }
    }
  })

  wrapper.trigger('click')
  expect(onClick).toHaveBeenCalled()
})

test('disabled is not trigger click event', () => {
  const onClick = jest.fn()
  const wrapper = mount(Button, {
    propsData: {
      disabled: true
    },
    context: {
      on: {
        click: onClick
      }
    }
  })

  wrapper.trigger('click')
  expect(onClick).toHaveBeenCalledTimes(0)
})
