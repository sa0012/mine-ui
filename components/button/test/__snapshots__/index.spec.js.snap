// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`size 1`] = `
<button class="ml-button ml-button--default ml-button--large ml-button">
  <div class="ml-button__content"><span class="ml-button__text"></span></div>
</button>
`;
